paper.md

```diff
@@ -0,0 +1,257 @@
+
+This extra lazy view is new functionality. 
+It would obviously have been possible to make the default not load the b-tree, but in the opinion of the
+current `pyfive` maintainers loading the b-tree at variable instantiation is likely more consistent
+with user expectations is it is more equivalent to the behaviour of other packages like `h5py`.
+
+## Cloud Optimisation
+
+To be fully cloud optimised - as defined by @Stern22 - an HDF5 file needs to have a contiguous index for each variable, 
+and the chunks for each variable need to be sensibly chosen and broadly contiguous within the file.
+When these criteria are met, indexes can be read efficiently, and middleware such as fsspec can make sensible use of readahead caching strategies.
+
+HDF5 data files direct from simulations and instruments are often not in this state as information about the number
+of variables, the number of chunks per variable, and the compressed size of those variables is not known as the data is being produced.  
+In such cases the data is also often not chunked along the dimensions being added to as the file is written (since it would have to be buffered first).
+
+Of course, once the file is produced, such information is available.
+Metadata can be repacked to the front of the file and variables can be rechunked and made continuous - 
+which is effectively the same process undertaken when HDF5 data is reformatted to other cloud optimised formats.
+
+The HDF5 library provides a tool "h5repack" which can do this, provided it is driven with suitable information about required chunk shape and the expected size of metadata fields. Versions (>1.0) of `pyfive` supports both a method to query whether such repacking is necessary, and to extract necessary parameters.
+
+In the following example we compare and contrast the unpacked and repacked version of a particularly pathological file, and in doing so howcase some of the `pyfive` API extensions which help us understand why it is pathological, and how to address those issues for repacking.
+
+If we extract just a piece of the output of `p5dump -s` on this file (which has surface wind velocity at three hour intervals for one hundred years):
+```
+float64 time(time) ;
+                time:standard_name = "time" ;
+                time:_n_chunks = 292192 ;
+                time:_chunk_shape = (1,) ;
+                time:_btree_range = (31808, 19854095942) ;
+                time:_first_chunk = 9094 ;
+
+float32 uas(time, lat, lon) ;
+                uas:_Storage = "Chunked" ;
+                uas:_n_chunks = 292192 ;
+                uas:_chunk_shape = (1, 143, 144) ;
+                uas:_btree_range = (28672, 19854809382) ;
+                uas:_first_chunk = 36520 ;
+```
+we can immediately see that this will be a problematic file!  The b-tree index is clearly interleaved with the data (compare the first chunk address with last index addresses of the two variables), and with a chunk dimension of (1,), any effort
```

Comment:
```suggestion
we can immediately see that this will be a problematic file! The internal file metadata index is clearly interleaved with the data (compare the first chunk address with last index addresses of the two variables), and with a chunk dimension of (1,), any effort
```